src/compiler/checker.ts

```diff
@@ -1427,8 +1438,9 @@ namespace ts {
 
                     switch (node.kind) {
                         case SyntaxKind.ArrowFunction:
-                        case SyntaxKind.PropertyDeclaration:
                             return true;
+                        case SyntaxKind.PropertyDeclaration:
+                            return stopAtAllPropertyDeclarations ? "quit": true;

```

Comment:
I can’t quite figure out what this means without stepping through one case that needs `stopAtAllPropertyDeclarations` as `true` and one that needs `false`, but are there any tests with a class expression as the initializer of a property declaration?

```ts
class Foo {
  Bar = class extends Foo {
    p2 = this.p1
  }
  p1 = 0
}
```

(I don’t know, maybe there’s a more creative combo of these horrors)

Reply:
There is one: 

```ts
class B2<V> {
    anon = class extends A<V> { }
}
```

Your example works on 3.7, so needs to keep working on 3.8:

https://www.typescriptlang.org/play/index.html#code/MYGwhgzhAECC0G8BQ1XQELQLzVJGApgB4AuBAdgCYzzJr3QAOATNtCQBYCWEAdIwEYUaAL7DUgtgAYkYgG5gATtDBtyBAO5wAFAEokwAPbkIhkAV4hDAc21h+A-QuVgARms0re6PQeOnzSxs7Vwd9IA

Back to the drawing board I guess. What a test case!

Reply:
Commit to fix this looks good 👍